app/routes.py

```diff
@@ -1,2 +1,297 @@
+from app.models.goal import Goal
+from app import db
+from datetime import datetime
+import requests
+import os
+from app.slackbot import GLaDOS
 
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+def err_404():
+    """Helper function for 404 errors"""
+    return "", 404
+
+
+#===============================TASKS===============================#
+
+
+def retrieve_tasks_ordered_by_title(is_desc=False):
+    """Helper function to get all tasks."""
+
+    current_tasks = []
+
+    if is_desc:
+        current_tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        current_tasks = Task.query.order_by(Task.title.asc()).all()
+
+    return [task.to_json_format() for task in current_tasks]
+        #: 'current_tasks' will store all tasks in the current session.
+        #: params: expects a boolean, that will return from the function
+        #: below -> is_sort_descending. If is_sort_descending returns
+        #: True, it will get the list of current tasks, and order those 
+        #: by title in descending order #[::-1]
+        #: else, orders titles by ascending order. returns a list of tasks
+
+
+def is_sort_descending(query_args):
+    """Helper function to verify the requested sorting order."""
+
+    if "sort" in query_args and query_args["sort"] == "asc":
+        return False
+
+    return True
+        #: helper function to check if the query string is asking 
+        #: for a specific sorting order (desc to be precise, hence
+        #: the name.) a GET request specifies that the sorting should
+        #: be *ascending*, returns False and this value will be passed
+        #: into 'retrieve_tasks_ordered_by_title' as argument. 
+        #: otherwise it returns True, meaning "yes, in descending order"
+        #: tried using the bytes returning from request.data, but by 
+        #: comparing byte strings I allowed any query 
+        #: with matching characters to be passed in the query.
+        #: "/?SORTeio_de_trapezio_DESCendente" was ordering by desc. 
+        #: if b"sort" in query_str and b"desc" in query_str.. 
+
+
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def get_all_tasks():
+    """Main GET request for tasks"""
+
+    return jsonify(retrieve_tasks_ordered_by_title(is_sort_descending(request.args)))
+        #: 1. jsonify() -> jsonifies
+        #: 2. retrieve_tasks_ordered_by_title -> gets all tasks (by title)
+        #: 3. is_sort_descending -> checks if an order is specifird
+        #: 4. request.args -> the arguments in the query, if any
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"], strict_slashes=False)
+def get_task(task_id):
+    """Selects a single task with specified ID"""
+
+    single_task = Task.query.get(task_id)
+
+    if not single_task:
+        return err_404()
+
+    return jsonify({"task": single_task.to_json_format()})
+
+
+def validate_field(field, dic):
```

Comment:
SOOOO COOOL! This is honestly another level of developer thinking, to abstract out the very idea of validation into one function like this!

Reply:
This could be taken even one step further by allowing an optional 3rd param for expected_type, which the function could use to verify that the type of the field is the type that's expected.